### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by ragangithub**
- Created a class Nameable
- Prepared base Decorator
- Prepared CapitalizeDecorator  
-  Prepared  TrimmerDecorator

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  #   Implement the correct_name method that returns the result of the correct_name method of the @nameable.
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (thecodechaser) [inline base_decorator.rb]: - [ ] Please make sure the `nameable` variable is readable and writable to work the decorator pattern as expected.
  - You can consider using `attr_accessor :nameable`

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super.capitalize
+  end
```
Comment (thecodechaser) [inline capitalize_decorator.rb]: - [ ] Please fix this method as below to work the CapitalizeDecorator as expected.

```ruby
def correct_name
      @nameable.correct_name.capitalize
    end
```

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    super[0..9]
+  end
```
Comment (thecodechaser) [inline trimmer_decorator.rb]: - [ ] Please fix this method as below to work the TrimmerDecorator as expected.

```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @ragangithub👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with the decorate pattern. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @ragangithub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Proper implementation of reviewer's suggestions

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**